app/src/main/java/org/breezyweather/main/MainActivity.kt

```diff
@@ -400,36 +412,54 @@ class MainActivity : GeoActivity(), HomeFragment.Callback, ManagementFragment.Ca
                 }
             }
         }
-        viewModel.snackbarError.observe(this) { error ->
-            if (error != null) {
-                val shortMessage = if (error.error == RefreshErrorType.FAILED_FEATURE) {
-                    getString(
-                        error.error.shortMessage,
-                        getString(error.feature!!.resourceName!!)
-                    )
-                } else {
-                    getString(error.error.shortMessage)
-                }
-                val shortMessageWithSource = if (!error.source.isNullOrEmpty()) {
-                    val sourceName = sourceManager.getSource(error.source)?.name ?: error.source
-                    "$sourceName${getString(
-                        R.string.colon_separator
-                    )}$shortMessage"
-                } else {
-                    shortMessage
+        viewModel.snackbarError.observe(this) { errors ->
+            if (errors.size > 1) {
+                SnackbarHelper.showSnackbar(
+                    content = getString(R.string.message_multiple_refresh_errors),
+                    action = getString(R.string.action_show)
+                ) {
+                    // Inefficient workaround to apply the correct theme while taking the location-based theme setting
+                    // into account. TODO: replace with a central solution which can be used in all composables
```

Comment:
This issue occurs every time a new composable is added to the MainActivity; at least that is my experience so far. I have some rough ideas in mind how to resolve this, so I'll create a new issue for this.